modules/ingest-common/src/main/java/org/opensearch/ingest/common/CommunityIDProcessor.java

```diff
@@ -0,0 +1,654 @@
+        String icmpTypeField,
+        String icmpCodeField,
+        int seed,
+        String targetField,
+        boolean ignoreMissing
+    ) {
+        super(tag, description);
+        this.sourceIPField = sourceIPField;
+        this.sourcePortField = sourcePortField;
+        this.destinationIPField = destinationIPField;
+        this.destinationPortField = destinationPortField;
+        this.ianaProtocolNumberField = ianaProtocolNumberField;
+        this.protocolField = protocolField;
+        this.icmpTypeField = icmpTypeField;
+        this.icmpCodeField = icmpCodeField;
+        this.seed = seed;
+        this.targetField = targetField;
+        this.ignoreMissing = ignoreMissing;
+    }
+
+    public String getSourceIPField() {
+        return sourceIPField;
+    }
+
+    public String getSourcePortField() {
+        return sourcePortField;
+    }
+
+    public String getDestinationIPField() {
+        return destinationIPField;
+    }
+
+    public String getDestinationPortField() {
+        return destinationPortField;
+    }
+
+    public String getIANAProtocolNumberField() {
+        return ianaProtocolNumberField;
+    }
+
+    public String getProtocolField() {
+        return protocolField;
+    }
+
+    public String getIcmpTypeField() {
+        return icmpTypeField;
+    }
+
+    public String getIcmpCodeField() {
+        return icmpCodeField;
+    }
+
+    public int getSeed() {
+        return seed;
+    }
+
+    public String getTargetField() {
+        return targetField;
+    }
+
+    public boolean isIgnoreMissing() {
+        return ignoreMissing;
+    }
+
+    @Override
+    public IngestDocument execute(IngestDocument document) {
+        // resolve protocol firstly
+        Protocol protocol = resolveProtocol(document);
+        // exit quietly if protocol cannot be resolved and ignore_missing is true
+        if (protocol == null) {
+            return document;
+        }
+
+        // resolve ip secondly, exit quietly if either source ip or destination ip cannot be resolved and ignore_missing is true
+        byte[] sourceIPByteArray = resolveIP(document, sourceIPField);
+        if (sourceIPByteArray == null) {
+            return document;
+        }
+        byte[] destIPByteArray = resolveIP(document, destinationIPField);
+        if (destIPByteArray == null) {
+            return document;
+        }
+        // source ip and destination ip must have same format, either ipv4 or ipv6
+        if (sourceIPByteArray.length != destIPByteArray.length) {
+            throw new IllegalArgumentException("source ip and destination ip must have same format");
+        }
+
+        // resolve source port and destination port for transport protocols,
+        // exit quietly if either source port or destination port is null nor empty
+        Integer sourcePort = null;
+        Integer destinationPort = null;
+        final boolean isTransportProtocol = Protocol.isTransportProtocol(protocol.getProtocolCode());
```

Comment:
Would be great to make it part of the enum:

```suggestion
        final boolean isTransportProtocol = protocol.isTransportProtocol();
```